db/config.cc

```diff
@@ -229,6 +230,11 @@ class convert<enum_option<db::tri_mode_restriction_t>> {
 #define str(x)  #x
 #define _mk_init(name, type, deflt, status, desc, ...)  , name(this, str(name), value_status::status, type(deflt), desc)
 
+static db::tri_mode_restriction_t::mode strict_allow_filtering_default() {
+    return scylla_version().starts_with("4.5") ?
```

Comment:
This now strikes me as over-engineered.  We can just have different defaults in branch-4.5 and master.

Reply:
This also doesn't handle 4.4 (also impacted by #7608) and Enterprise versions.

Reply:
Certainly the code itself shouldn't refer to versions. Also why would the defaults be different? We need to allow for a long grace period.